src/components/Microsoft.DotNet.UpgradeAssistant.MSBuild/MSBuildProject.NuGetPackages.cs

```diff
@@ -68,38 +60,14 @@ public IEnumerable<NuGetReference> PackageReferences
             }
         }
 
-        public IEnumerable<NuGetReference> GetTransitivePackageReferences(TargetFrameworkMoniker tfm)
-        {
-            if (PackageReferenceFormat != NugetPackageFormat.PackageReference)
-            {
-                return Enumerable.Empty<NuGetReference>();
-            }
-
-            var parsedTfm = NuGetFramework.Parse(tfm.Name);
-            var lockFile = LockFileUtilities.GetLockFile(LockFilePath, NuGet.Common.NullLogger.Instance);
-
-            if (lockFile is null)
-            {
-                throw new ProjectRestoreRequiredException($"Project is not restored: {FileInfo}");
-            }
+        public IAsyncEnumerable<NuGetReference> GetTransitivePackageReferencesAsync(TargetFrameworkMoniker tfm, CancellationToken token)
+            => GetAllDependenciesAsync(tfm, token).Select(l => new NuGetReference(l.Name, l.Version.ToNormalizedString()));
 
-            var lockFileTarget = lockFile
-                .Targets
-                .FirstOrDefault(t => t.TargetFramework.DotNetFrameworkName.Equals(parsedTfm.DotNetFrameworkName, StringComparison.Ordinal));
+        public ValueTask<bool> IsTransitivelyAvailableAsync(string packageName, CancellationToken token)
+            => TargetFrameworks.ToAsyncEnumerable().AnyAwaitAsync(tfm => ContainsDependencyAsync(tfm, d => string.Equals(packageName, d.Id, StringComparison.OrdinalIgnoreCase), token), cancellationToken: token);
 
-            if (lockFileTarget is null)
-            {
-                throw new ProjectRestoreRequiredException($"Could not find {parsedTfm.DotNetFrameworkName} in {LockFilePath} for {FileInfo}");
-            }
-
-            return lockFileTarget.Libraries.Select(l => new NuGetReference(l.Name, l.Version.ToNormalizedString()));
-        }
-
-        public bool IsTransitivelyAvailable(string packageName)
-            => TargetFrameworks.Any(tfm => ContainsDependency(tfm, d => string.Equals(packageName, d.Id, StringComparison.OrdinalIgnoreCase)));
-
-        public bool IsTransitiveDependency(NuGetReference nugetReference)
-            => TargetFrameworks.Any(tfm => ContainsDependency(tfm, d => ReferenceSatisfiesDependency(d, nugetReference, true)));
+        public ValueTask<bool> IsTransitiveDependencyAsync(NuGetReference nugetReference, CancellationToken token)
```

Comment:
I'd like to see more method level comments in this file to help keep me out of reading the code to figure out our intent.

I think the intent here is to build a helper function that returns true if the NuGetReference is transitively available. What I don't understand is the context. Are we saying transitively available because of an existing NuGet package, or maybe available as a dependency in the context of a project? If it's a project context, I guess I assume it's the currently selected project.

Reply:
It's yes to both, but only the ones that have been set as not private. I'll add some comments to the interface